[2025-04-22T13:26:20.476+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-04-22T13:26:20.500+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_kafka_pipeline.Kafka_Consumer manual__2025-04-22T13:25:20.691673+00:00 [queued]>
[2025-04-22T13:26:20.510+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_kafka_pipeline.Kafka_Consumer manual__2025-04-22T13:25:20.691673+00:00 [queued]>
[2025-04-22T13:26:20.510+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-04-22T13:26:20.526+0000] {taskinstance.py:2330} INFO - Executing <Task(BashOperator): Kafka_Consumer> on 2025-04-22 13:25:20.691673+00:00
[2025-04-22T13:26:20.534+0000] {standard_task_runner.py:63} INFO - Started process 102 to run task
[2025-04-22T13:26:20.536+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'stock_kafka_pipeline', 'Kafka_Consumer', 'manual__2025-04-22T13:25:20.691673+00:00', '--job-id', '37', '--raw', '--subdir', 'DAGS_FOLDER/stock_kafka_pipeline.py', '--cfg-path', '/tmp/tmpmvfvf66e']
[2025-04-22T13:26:20.537+0000] {standard_task_runner.py:91} INFO - Job 37: Subtask Kafka_Consumer
[2025-04-22T13:26:20.565+0000] {task_command.py:426} INFO - Running <TaskInstance: stock_kafka_pipeline.Kafka_Consumer manual__2025-04-22T13:25:20.691673+00:00 [running]> on host c25f8f7dc196
[2025-04-22T13:26:20.627+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='stock_kafka_pipeline' AIRFLOW_CTX_TASK_ID='Kafka_Consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-04-22T13:25:20.691673+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-22T13:25:20.691673+00:00'
[2025-04-22T13:26:20.627+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-04-22T13:26:20.643+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-04-22T13:26:20.644+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'python3 /opt/airflow/Scripts/load_to_bronze.py']
[2025-04-22T13:26:20.656+0000] {subprocess.py:86} INFO - Output:
[2025-04-22T13:26:21.445+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection client_id=kafka-python-2.1.5, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.27.0.6', 9092)]>: connecting to kafka:9092 [('172.27.0.6', 9092) IPv4]
[2025-04-22T13:26:21.448+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection client_id=kafka-python-2.1.5, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.27.0.6', 9092)]>: Broker version identified as 2.6
[2025-04-22T13:26:21.448+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection client_id=kafka-python-2.1.5, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.27.0.6', 9092)]>: Connection complete.
[2025-04-22T13:26:21.449+0000] {subprocess.py:93} INFO - INFO:kafka.consumer.subscription_state:Updating subscribed topics to: ('stock-prices',)
[2025-04-22T13:26:21.449+0000] {subprocess.py:93} INFO - INFO:__main__:Listening to Kafka topic 'stock-prices' and writing to /opt/airflow/includes/local warehose/bronze
[2025-04-22T13:26:21.452+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection client_id=kafka-python-2.1.5, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.27.0.6', 9092)]>: connecting to kafka:9092 [('172.27.0.6', 9092) IPv4]
[2025-04-22T13:26:21.452+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection client_id=kafka-python-2.1.5, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.27.0.6', 9092)]>: Connection complete.
[2025-04-22T13:26:21.453+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection client_id=kafka-python-2.1.5, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.27.0.6', 9092)]>: Closing connection.
[2025-04-22T13:26:21.555+0000] {subprocess.py:93} INFO - INFO:kafka.cluster:Group coordinator for stock-consumer-group is BrokerMetadata(nodeId='coordinator-1', host='kafka', port=9092, rack=None)
[2025-04-22T13:26:21.555+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Discovered coordinator coordinator-1 for group stock-consumer-group
[2025-04-22T13:26:21.555+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Starting new heartbeat thread
[2025-04-22T13:26:21.555+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator.consumer:Revoking previously assigned partitions set() for group stock-consumer-group
[2025-04-22T13:26:21.556+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection client_id=kafka-python-2.1.5, node_id=coordinator-1 host=kafka:9092 <connecting> [IPv4 ('172.27.0.6', 9092)]>: connecting to kafka:9092 [('172.27.0.6', 9092) IPv4]
[2025-04-22T13:26:21.556+0000] {subprocess.py:93} INFO - INFO:kafka.conn:<BrokerConnection client_id=kafka-python-2.1.5, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.27.0.6', 9092)]>: Connection complete.
[2025-04-22T13:26:21.657+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:(Re-)joining group stock-consumer-group
[2025-04-22T13:26:24.661+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Elected group leader -- performing partition assignments using range
[2025-04-22T13:26:24.666+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator:Successfully joined group stock-consumer-group with generation 7
[2025-04-22T13:26:24.666+0000] {subprocess.py:93} INFO - INFO:kafka.consumer.subscription_state:Updated partition assignment: [TopicPartition(topic='stock-prices', partition=0)]
[2025-04-22T13:26:24.666+0000] {subprocess.py:93} INFO - INFO:kafka.coordinator.consumer:Setting newly assigned partitions {TopicPartition(topic='stock-prices', partition=0)} for group stock-consumer-group
[2025-04-22T13:26:24.676+0000] {subprocess.py:93} INFO - INFO:__main__:Wrote /opt/airflow/includes/local warehose/bronze/IBM_1745328326.json
[2025-04-22T13:26:24.676+0000] {subprocess.py:93} INFO - INFO:__main__:Wrote /opt/airflow/includes/local warehose/bronze/AAPL_1745328338.json
[2025-04-22T13:26:24.676+0000] {subprocess.py:93} INFO - INFO:__main__:Wrote /opt/airflow/includes/local warehose/bronze/MSFT_1745328351.json
[2025-04-22T13:26:24.676+0000] {subprocess.py:93} INFO - INFO:__main__:Wrote /opt/airflow/includes/local warehose/bronze/TSLA_1745328363.json
[2025-04-22T13:26:31.578+0000] {local_task_job_runner.py:124} ERROR - Received SIGTERM. Terminating subprocesses
[2025-04-22T13:26:31.579+0000] {process_utils.py:132} INFO - Sending 15 to group 102. PIDs of all processes in the group: [103, 102]
[2025-04-22T13:26:31.580+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 102
[2025-04-22T13:26:31.587+0000] {taskinstance.py:2611} ERROR - Received SIGTERM. Terminating subprocesses.
[2025-04-22T13:26:31.588+0000] {subprocess.py:104} INFO - Sending SIGTERM signal to process group
[2025-04-22T13:26:31.588+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-04-22T13:26:31.620+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 234, in execute
    result = self.subprocess_hook.run_command(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/hooks/subprocess.py", line 91, in run_command
    for raw_line in iter(self.sub_process.stdout.readline, b""):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2613, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2025-04-22T13:26:31.635+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=stock_kafka_pipeline, task_id=Kafka_Consumer, run_id=manual__2025-04-22T13:25:20.691673+00:00, execution_date=20250422T132520, start_date=20250422T132620, end_date=20250422T132631
[2025-04-22T13:26:31.687+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=102, status='terminated', exitcode=2, started='13:26:20') (102) terminated with exit code 2
[2025-04-22T13:26:31.688+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=103, status='terminated', started='13:26:20') (103) terminated with exit code None
[2025-04-22T13:26:31.691+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 143
[2025-04-22T13:26:31.745+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
